cli/command/container/utils.go

```diff
@@ -38,6 +38,7 @@ func waitExitOrRemoved(ctx context.Context, apiClient client.APIClient, containe
 	go func() {
 		select {
 		case <-ctx.Done():
+			close(statusC)
```

Comment:
Should we define a status code for these scenarios where the ctx has been canceled? Could it be useful to send that along before closing the channel?

Reply:
I think we should keep the previous behaviour here, which I think this patch does (check the added e2e tests). I tested locally, and the old exit code was 127, and with this patch it's still 127. In the container fails to start+autoremove code path, we don't check the result from that channel (see https://github.com/docker/cli/blob/05cba3f7b97c73d49675a3e5adb1118fe0c79867/cli/command/container/run.go#L204).

Reply:
The only case that could cause issues is https://github.com/docker/cli/blob/05cba3f7b97c73d49675a3e5adb1118fe0c79867/cli/command/container/run.go#L234, maybe we should send there? I'm not sure if we'd ever hit that line with a context cancelled, but I guess it shouldn't hurt.

Reply:
Changed this up more, we shouldn't even be getting here since there's no reason to cancel the context if `--rm` is set.